cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server_test.go

```diff
@@ -404,3 +404,223 @@ func TestGetAvailablePackageSummaries(t *testing.T) {
 		})
 	}
 }
+
+func TestAvailablePackageDetailFromChart(t *testing.T) {
+	testCases := []struct {
+		name       string
+		chart      *models.Chart
+		expected   *corev1.AvailablePackageDetail
+		statusCode codes.Code
+	}{
+		{
+			name:       "it returns AvailablePackageDetail if the chart is correct",
+			chart:      chartOK,
+			expected:   availablePackageDetailOK,
+			statusCode: codes.OK,
+		},
+		{
+			name:       "it returns internal error if empty chart",
+			chart:      &models.Chart{},
+			statusCode: codes.Internal,
+		},
+		{
+			name:       "it returns internal error if chart is invalid",
+			chart:      &models.Chart{Name: "foo"},
+			statusCode: codes.Internal,
+		},
```

Comment:
I guess you could add failing tests here to show that it should succeed without certain fields (ie. readme, icon, maintainers) then fix it for them to pass :)

Reply:
TDD is great for these fixes :P Added a `TestIsValidChart` test suite